modules/editor-service/src/main/java/org/opencastproject/editor/EditorServiceImpl.java

```diff
@@ -0,0 +1,779 @@
+  private static final String DEFAULT_SMIL_CATALOG_TAGS = "archive";
+  private static final String DEFAULT_SMIL_SILENCE_FLAVOR = "*/silence";
+  private static final String DEFAULT_PREVIEW_VIDEO_SUBTYPE = "video+preview";
+  private static final String DEFAULT_PREVIEW_AUDIO_SUBTYPE = "audio+preview";
+
+  public static final String OPT_PREVIEW_SUBTYPE = "preview.subtype";
+  public static final String OPT_PREVIEW_TAG = "preview.tag";
+  public static final String OPT_SMIL_CATALOG_FLAVOR = "smil.catalog.flavor";
+  public static final String OPT_SMIL_CATALOG_TAGS = "smil.catalog.tags";
+  public static final String OPT_SMIL_SILENCE_FLAVOR = "smil.silence.flavor";
+  public static final String OPT_PREVIEW_AUDIO_SUBTYPE = "preview.audio.subtype";
+  public static final String OPT_PREVIEW_VIDEO_SUBTYPE = "preview.video.subtype";
+
+  private Set<String> smilCatalogTagSet = new HashSet<>();
+
+  void setSecurityService(SecurityService securityService) {
+    this.securityService = securityService;
+  }
+
+  void setSmilService(SmilService smilService) {
+    this.smilService = smilService;
+  }
+
+  void setWorkflowService(WorkflowService workflowService) {
+    this.workflowService = workflowService;
+  }
+
+  void setWorkspace(Workspace workspace) {
+    this.workspace = workspace;
+  }
+
+  void setUrlSigningService(UrlSigningService urlSigningService) {
+    this.urlSigningService = urlSigningService;
+  }
+
+  void setAssetManager(AssetManager assetManager) {
+    this.assetManager = assetManager;
+  }
+
+  public void setAdminUISearchIndex(AdminUISearchIndex adminUISearchIndex) {
+    this.searchIndex = adminUISearchIndex;
+  }
+
+  public void setIndexService(IndexService index) {
+    this.index = index;
+  }
+
+  public MediaPackageElementFlavor getSmilCatalogFlavor() {
+    return smilCatalogFlavor;
+  }
+
+  public Set<String> getSmilCatalogTags() {
+    return smilCatalogTagSet;
+  }
+
+  public String getPreviewVideoSubtype() {
+    return previewVideoSubtype;
+  }
+
+  public MediaPackageElementFlavor getSmilSilenceFlavor() {
+    return smilSilenceFlavor;
+  }
+
+  private String getPreviewSubtype() {
+    return previewSubtype;
+  }
+
+  public String getPreviewTag() {
+    return previewTag;
+  }
+
+  @Activate
+  @Modified
+  public void activate(ComponentContext cc) {
+    Dictionary<String, Object> properties = cc.getProperties();
+    expireSeconds =  UrlSigningServiceOsgiUtil.getUpdatedSigningExpiration(properties, this.getClass().getSimpleName());
+    signWithClientIP = UrlSigningServiceOsgiUtil.getUpdatedSignWithClientIP(properties,this.getClass().getSimpleName());
+    // Preview tag
+    previewTag = StringUtils
+            .defaultString((String) properties.get(OPT_PREVIEW_TAG), DEFAULT_PREVIEW_TAG);
+    logger.debug("Preview subtype configuration set to '{}'", previewTag);
+
+    // Preview subtype
+    previewSubtype = StringUtils.defaultString((String) properties.get(OPT_PREVIEW_SUBTYPE), DEFAULT_PREVIEW_SUBTYPE);
```

Comment:
```suggestion
    previewSubtype = Objects.toString(properties.get(OPT_PREVIEW_SUBTYPE), DEFAULT_PREVIEW_SUBTYPE);
```

Same below. `Objects.toString(Object, defaultString)` is actually a bit simpler for these cases.

Reply:
done